### DISCUSSION TITLE
Pass response as argument with generics

### DISCUSSION THREAD

**Discussion by wenerme** [connectrpc/connect-go#262] [Q&A] [answered]
Every function has a line of `&connect_go.Response[svcv1.MethodResponse]{}`, this is even harder to write with generic, if pass response as argument, will be easier to write, since connect-go is new, I think this is possible, I see this in go-micro, this makes DX alot better.

https://github.com/asim/go-micro/blob/1cd7cfaa6cbb7d4cd9cba76ae2374469649a81d4/plugins/server/grpc/extractor_test.go#L17-L19

**Comment by bufdev** [comment, marked as answer]
I think you just mean to use:

- https://pkg.go.dev/github.com/bufbuild/connect-go#NewRequest
- https://pkg.go.dev/github.com/bufbuild/connect-go#NewResponse

Examples are at:

- https://github.com/bufbuild/connect-go/blob/9f157f27e1000767c74a89ad6b8dfb47e8737e8f/client_example_test.go#L44
- https://github.com/bufbuild/connect-go/blob/9f157f27e1000767c74a89ad6b8dfb47e8737e8f/handler_example_test.go#L37

Is this what you are looking for?
